src/twisted/test/test_twistd.py

```diff
@@ -1006,29 +1008,25 @@ def test_setUidWithoutGid(self):
         with a UID and no GUID will result in the GUID being
         set to the default GUID for that UID.
         """
-        self._setUID("foo", 5151, "bar", 4242)
+        self._setUID("foo", 5151, "bar", 4242, "test_setUidWithoutGid.pid")
```

Comment:
Using a fixed path name seems like a recipe for littering the filesystem and potential unnecessary cross-process conflicts.  As long as we're fixing it, let's fix it so it's consistent with our test idiom for filesystem stuff:
```suggestion
        self._setUID("foo", 5151, "bar", 4242, self.mktemp() + "_test_setUidWithoutGid.pid")
```